### PR TITLE
[CPU] Fix mixing VEX and non-VEX instructions

### PR DESCRIPTION
**Pull request by EgorDuplensky**
### Details:
 -  Quote:
    The Skylake microarchitecture implements a different state
    machine than prior generations to manage the YMM state transition
    associated with mixing SSE and AVX instructions.
    It no longer saves the entire upper YMM state when executing
    an SSE instruction when in “Modified and Unsaved” state,
    but saves the upper bits of individual register.
    As a result, mixing SSE and AVX instructions will experience
    a penalty associated with partial register dependency of
    the destination registers being used and additional blend
    operation on the upper bits of the destination registers.
    
    Such type of penalties have a huge impact on openvino's and oneDNN's kernels.
    Basically the mixing of VEX and non-VEX instructions should be avoided.

- openvino oneDNN PR:
  https://github.com/openvinotoolkit/oneDNN/pull/75

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang Please take a look. There are some additional changes required above your PR to fix related perf issues.

**Comment by chenhu-wang** [comment]
Typically we create avx2 kernel on avx2 available target. On target w/ avx, but w/o avx2. we still create sse41 kernel. In this case,  all the branch of if (isa == cpu::x64::sse41) {} need use uni_xxx version. Some non-vex instructions in this branch will be mixed with uni_xxx instruction, which will use vex version on avx. Another possible is we always create sse41 kernel for all targets because some implementation limit like don't allow cross lane operation to reduce implementation complexity.
I find lots of more "if (isa == cpu::x64::sse41)"  code pass need to updated to use uni_xxx instruction version. 
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/emitters/jit_eltwise_emitters.cpp#L711
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/emitters/jit_eltwise_emitters.cpp#L650
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/emitters/jit_eltwise_emitters.cpp#L772
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/emitters/jit_eltwise_emitters.cpp#L833
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/emitters/jit_eltwise_emitters.cpp#L894
and more in jit_eltwise_emitters.cpp
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp#L368
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp#L411
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp#L515
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp#L567
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_pooling_node.cpp#L174

**Comment by EgorDuplensky** [comment]
@chenhu-wang 
`Typically we create avx2 kernel on avx2 available target. On target w/ avx, but w/o avx2. we still create sse41 kernel. In this case, all the branch of if (isa == cpu::x64::sse41) {} need use uni_xxx version`
Thank you for this comment.
I was also thinking about that, and the question is: do we have any Skylake CPUs (or newer) which only supports AVX and does not support AVX2, considering the fact that this problem appeared only since Skylake arch.
But, anyway, since we are going to implement basically the same change for the master, i agree that for master branch it would be better to update those sse41 branches with uni_* methods, just to align implementation.

**Comment by dmitry-gorokhov** [comment]
> @chenhu-wang
> `Typically we create avx2 kernel on avx2 available target. On target w/ avx, but w/o avx2. we still create sse41 kernel. In this case, all the branch of if (isa == cpu::x64::sse41) {} need use uni_xxx version`
> Thank you for this comment.
> I was also thinking about that, and the question is: do we have any Skylake CPUs (or newer) which only supports AVX and does not support AVX2, considering the fact that this problem appeared only since Skylake arch.
> But, anyway, since we are going to implement basically the same change for the master, i agree that for master branch it would be better to update those sse41 branches with uni_* methods, just to align implementation.

@EgorDuplensky  @chenhu-wang 

1) I am not aware about such HW. Anyway lets consider non avx2 cases separatelly
2) Agree. Lets update sse codepaths on uni_* instructions

**Comment by chenhu-wang** [comment]
> > @chenhu-wang
> > `Typically we create avx2 kernel on avx2 available target. On target w/ avx, but w/o avx2. we still create sse41 kernel. In this case, all the branch of if (isa == cpu::x64::sse41) {} need use uni_xxx version`
> > Thank you for this comment.
> > I was also thinking about that, and the question is: do we have any Skylake CPUs (or newer) which only supports AVX and does not support AVX2, considering the fact that this problem appeared only since Skylake arch.
> > But, anyway, since we are going to implement basically the same change for the master, i agree that for master branch it would be better to update those sse41 branches with uni_* methods, just to align implementation.
> 
> @EgorDuplensky @chenhu-wang
> 
> 1. I am not aware about such HW. Anyway lets consider non avx2 cases separatelly
> 2. Agree. Lets update sse codepaths on uni_* instructions

Mixed vex and non-vex instructions always have perf penalty for states transition, no matter it was on Skylake or other target such as Sandy Bridge. Skylake just have a different transition manner than prior and have different degrees of penalty. So I think it make senses to update sse codepath to uni_* if we concern perf on older targets in separate PR.  Current changes looks good for target with avx2.

**Review by chenhu-wang — APPROVED** [review]

**Comment by EgorDuplensky** [comment]
> > > @chenhu-wang
> > > `Typically we create avx2 kernel on avx2 available target. On target w/ avx, but w/o avx2. we still create sse41 kernel. In this case, all the branch of if (isa == cpu::x64::sse41) {} need use uni_xxx version`
> > > Thank you for this comment.
> > > I was also thinking about that, and the question is: do we have any Skylake CPUs (or newer) which only supports AVX and does not support AVX2, considering the fact that this problem appeared only since Skylake arch.
> > > But, anyway, since we are going to implement basically the same change for the master, i agree that for master branch it would be better to update those sse41 branches with uni_* methods, just to align implementation.
> > 
> > 
> > @EgorDuplensky @chenhu-wang
> > 
> > 1. I am not aware about such HW. Anyway lets consider non avx2 cases separatelly
> > 2. Agree. Lets update sse codepaths on uni_* instructions
> 
> Mixed vex and non-vex instructions always have perf penalty for states transition, no matter it was on Skylake or other target such as Sandy Bridge. Skylake just have a different transition manner than prior and have different degrees of penalty. So I think it make senses to update sse codepath to uni_* if we concern perf on older targets in separate PR. Current changes looks good for target with avx2.

**Comment by EgorDuplensky** [comment]
@dmitry-gorokhov 
CI is passed, i guess we can merge it
